shell/platform/fuchsia/dart/BUILD.gn

```diff
@@ -124,7 +124,7 @@ dart_library("args") {
   package_root = "$dart_src/third_party/pkg/args"
   package_name = "args"
 
-  pubspec = "$package_root/pubspec.yaml"
+  language_version = "2.17"
```

Comment:
I'm curious about this change. The upstream dart sdk CL brought in a version of package:args that now has an sdk requirement of `^3.0.0`. Previously it had been `'>=2.19.0 <4.0.0'`.

Is this file generally auto-maintained / generated as package metadata changes? Manually maintained w/ duplicate info? Should it be parsing the required sdk version out of the pubspec?

Reply:
This is a Fuchsia-specific build script for package:args.

I don't know its history, but AFAIK it's manually maintained.

Reply:
Gotcha; thanks for the fix!

It looks like the parser does not handle the caret syntax for sdk constraints (`^3.0.0`):https://github.com/flutter/engine/blob/main/tools/fuchsia/dart/gen_dart_package_config.py#L55

And, the script defaults to Dart 2.8, which is super old. 3.0 might be a better default? Or 2.12? Or no default at all, and just failing, which may be more predictable.